### DISCUSSION TITLE
2022-01-10 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#962] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Frontend update](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932002)
- [Backend update](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932015)
- [Design update](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932072)
- [Product update](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932149)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932153)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1936426)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1940120)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1941019)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1941379)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1941983)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1942023)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1943482)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Frontend update

### Completed
1. PRs merged last week:
    1. #857 
    2. #931
2. #915 is out of draft and being reviewed
3. Work is progressing on #926

### Current Priorities
1. Review and merge PRs ready for review. 
    1. #826
    2. #861 
    3. #870 
    4. #915
    5. #961
2. Get #926 ready to review
3. Continue to make progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone. There only seem to be two unblocked issues.
    1. #500 
    2. #531
4. Work on some of the [14 unassigned frontend issues](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+label%3A%22work%3A+frontend%22+milestone%3A%22[07.1]+2022-01+improvements%22+no%3Aassignee) in the [2022-01 improvements milestone](https://github.com/centerofci/mathesar/milestone/33) while we're waiting for the type-related frontend PRs to be merged and more backend work to be done on data types. Feel free to also move some into later milestones if they're not critical.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good

**Reply by pavish** [reply]
Looks good to me.

I'll move some of the unassigned issues in the 2022-01 improvements milestone to later milestones. Most of them are not critical and good first issues for new contributors.

**Comment by kgodey** [comment]
## Backend update

### Completed

1. #865 is now merged, which means we have DB support for all date and time types!
2. Two of the three issues blocking the frontend are merged:
    1. #932 
    2. #933
3. Progress has been made on the following PRs:
    1. #921 
    2. #909 
    3. #797

### Current priorities

1. API error handling issues:
    1.  #859 (top priority since it's blocking the frontend)
    2. #623
    3. #883 
    4. #889
2. Finalize the approach to filtering (please prioritize responding to @dmos62's requests promptly) and get these draft PRs closer to merge:
    1. #921 
    2. #879 
    3. #893
3. Complete these two open PRs from a contributor and get them merged.
    1. #797 (assigned to @mathemancer and already reviewed once last week)
    2. #817 
4. Get this draft PR closer to merge: #885 
5. Continue to make progress on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)  issues.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by dmos62** [reply]
Concerning 2 (#921), prompt feedback thoughout the week would be great. When I was doing the previous revision of this refactor, I asked for feedback infrequently, which ended up being time-inefficient. I'm now trying to ask for feedback more often.

**Reply by mathemancer** [reply]
I've read this a few times, and I can't tell which is the third issue blocking the front end.  (I'm sure I'm missing something obvious)  Would you mind pointing it out, @kgodey ?

**Reply by kgodey** [reply]
@mathemancer it's #859, I folded it into the API error handling bucket since it's related to it and that's also the top priority this week.

**Reply by silentninja** [reply]
Sorry for taking so long with the error handing, I hope to finish off the error handling issues in the next few days. Other issues make sense.

**Reply by mathemancer** [reply]
@kgodey Ah, okay thanks; that makes sense.

**Comment by kgodey** [comment]
## Design update

### Completed
1. Second round of review on https://github.com/centerofci/mathesar-wiki/pull/16
2. Review opened on new spec: https://github.com/centerofci/mathesar-wiki/pull/19
3. Some work on the Mathesar-related portion of CCI's 2020/2021 Impact Report

### Current priorities 
1. Design work on #960 
2. Get currently open specs reviewed and hopefully merged:
    1. https://github.com/centerofci/mathesar-wiki/pull/16
    2. https://github.com/centerofci/mathesar-wiki/pull/19
3. Create first draft of spec for #466 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ppii775** [reply]
Sounds good to me! @ghislaineguerin Feel free to reach out and let me know how I can help!

**Comment by kgodey** [comment]
## Product update

### Completed
1. Definition of Mathesar Types and related discussions:
    1. #959 
    2. The [Data Types "Concepts" wiki page](https://wiki.mathesar.org/product/concepts/data-types) 
    3. The [Mathesar Types architecture wiki page](https://wiki.mathesar.org/en/engineering/architecture/mathesar-types)
2. Other wiki pages for [product concepts](https://wiki.mathesar.org/en/product/concepts):
    1. [Databases](https://wiki.mathesar.org/product/concepts/databases)
    2. [Schemas](https://wiki.mathesar.org/product/concepts/schemas)
    3. [Tables](https://wiki.mathesar.org/product/concepts/tables)
    4. [Views](https://wiki.mathesar.org/product/concepts/views)
3. Tickets created or updated for completed design specs:
    1. https://github.com/centerofci/mathesar-wiki/pull/12
        1. #257 
        2. #929 
        3. #930 
        4. #936 
        5. #939 
        6. #927 
        7. #928  
    2. https://github.com/centerofci/mathesar-wiki/pull/13:
        1. #942
        1. #943
        1. #944
        1. #945
        1. #946
        1. #947
        1. #948
        1. #949
        1. #951
        1. #950
        1. #952
        1. #953
        1. #955
        1. #954

### Current priorities
1. Finalize newly created issues listed above by reviewing and resolving any questions/discussion.
2. Define data sources in Views better and update other Views tickets: #806
3. Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-01-10 to 2022-01-14 | Brent |
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |
| 2022-01-31 to 2022-02-04 | Mukesh |
| 2022-02-07 to 2022-02-11 | Sean |
| 2022-02-14 to 2022-02-18 | Dominykas |
| 2022-02-21 to 2022-02-25 | Ghislaine |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by dmos62** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I think it's a good idea. Seeing what we've done during last week gives me a sense of pace.

>  How did your most recent work go?

I spent last week mostly rethinking the filtering refactor. I'm pretty happy with where it's at, though it's still in draft. I also participated in some architectural discussions, which was interesting because everyone had different ideas.

> What are you planning to work on soon?

I'm currently in the feedback-iteration loop with the filtering refactor. I expect it to be my principal task next week.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No, except that prompt feedback on the filtering refactor would be useful.

**Comment by seancolsen** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

- It doesn't add any new value for me personally, so if it's taking too much time, then I think it's fine to cut for the sake of prioritizing other work. But if other people benefit from it, then that's great!

> How did your most recent work go?

- Last week was all tickets and PR review for me. No actual coding. That's fine though.
- I feel pretty good about all the tickets and product-related work I did early in the week and would be happy to continue pitching in with that stuff as needed. I enjoy thinking through the subtle edge cases, implementation details, and UX considerations. Plus, having gotten my head so deeply into one topic, I'm well positioned for future development on those features.

> What are you planning to work on soon?

- I'm still wrapping up my review of the Form Builder PR. Pavish spent a long time creating the PR, so I'm giving it an appropriately thorough review. Hoping to be done with that soon though. Ideally today.
- I need to get back to #531 but it's been a while since I last worked on it! That's my next task. There's a tricky question about modal UX and implementation that Pavish and I discussed before holiday. I'll need to work on resolving that before I can make more progress on the underlying feature. Not sure how long that'll take.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

> Personal updates

- Last Tuesday, our kids' school opened back up from winter break. However, due to the Omicron surge, we chose to keep the kids home for safety's sake. Now both of their teachers have covid symptoms and their class is closed today. They're still waiting on test results which apparently are quite backed up. We're feeling good about our decision last week, but it's also been hard having the kids home, which is why some of my days were shorter last week. It looks like we'll probably have the kids home all week this week too. Really hoping this surge dies down soon!

**Reply by kgodey** [reply]
> I feel pretty good about all the tickets and product-related work I did early in the week and would be happy to continue pitching in with that stuff as needed.

I would be happy for your help with this, I think you're doing an excellent job with the tickets.

---

Let us know if we can support you in any way while your kids are home.

**Comment by kgodey** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I think it's nice to see what we've been accomplishing, but I don't want to spend time on it if it's useless. 

> How did your most recent work go?

It was nice to finally get some product work done, and to get @seancolsen's help with the tickets. I'm happy with where we've ended up with Mathesar Data Types especially.

> What are you planning to work on soon?

-  Finalize newly created issues by reviewing and resolving any questions/discussion.
-  Define data sources in Views better and update other Views tickets: #806
- Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853 
- I'll probably schedule a team "State of Mathesar" check in this week as well
- I also need to figure out a plan for any hires we want to make soon

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nothing in particular.

**Comment by pavish** [comment]
## Pavish's check in

> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I usually take a look at all newly created issues, discussions and PRs that are merged. I don't see much value in including this, but I still think it'll be good to use this as a weekly record to track progress.

> How did your most recent work go?

Last week was a bit slow but I was more focused on analyzing and writing code, which felt good. Form builder is ready and type config PR needs a few more tweaks. I also started working on CellInput component.

> What are you planning to work on soon?

* Get Form builder PR merged.
* Get Type config PR ready and possibly merged.
* Raise discussions for APIs for Database and Display options in type selection menu.
* Raise draft PR for CellInput.
* I also intend to focus more on reviewing open PRs this week.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None.

**Reply by kgodey** [reply]
@pavish What issue is the CellInput component related to?

**Reply by pavish** [reply]
@kgodey There isn't a specific issue for it. It is related to all the issues in Data types milestone and it is required for setting the default value of a column.

Each column type has different inputs to show on the table along with different validation, display and formatting options i.e. Boolean types need to show checkbox on the spreadsheet, number types have to only accept numbers etc., The CellInput component encapsulates all of that.

**Reply by kgodey** [reply]
Makes sense, thanks for the explanation. It would be to make issues for components you're working on (like #750 or #824) so it's easier for everyone to track work.

**Comment by ghislaineguerin** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I guess for design, and it doesn't change much as work is more long-lived. I still think it's helpful to keep a log.

> How did your most recent work go?

I think it went well. Specs are becoming more precise and easier to write. The prototype has many pieces I need to build the designs.

> What are you planning to work on soon?

1. Merge current open specs
2. Create the first draft of spec for #466 
3. Design work on #960 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- None

**Reply by kgodey** [reply]
@ghislaineguerin Your planned work doesn't seem similar to [the design priorities for this week](https://github.com/centerofci/mathesar/discussions/962#discussioncomment-1932072). Can you prioritize those?

If you have more time after this week's priorities and https://github.com/centerofci/mathesar-wiki/issues/18, I think it's better to wait on #781 until I'm done with product work on #806. I don't think we can fully define view creation until how we represent view data sources is fully defined.

**Comment by mathemancer** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I find it motivating, personally.

> How did your most recent work go?

It went well; I'm trying to focus on smaller, higher-volume tasks, and I think it's going alright.  As such, I got a few small PRs submitted and merged.

> What are you planning to work on soon?

My top priority is supporting @dmos62 's work on filtering if there's anything I can do there (reviewing PRs / ideas, taking up menial implementations of various details, etc.).  In particular, I had an idea for wrapping operators and setting up a function to reflect all needed info that might help, but will involve lots of ugly SQL (for the reflection, at least).   If @dmos62 thinks that would be useful, I could throw together a quick, independent PR that would add some examples of such wrapper functions to the DB, and the reflection function.  (If you don't think it'll help, @dmos62, just let me know.)  I'm also on comms.  Finally, I want to look through https://github.com/centerofci/mathesar/milestone/11 for any small(ish) issues to pick up and knock out.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

The grouping and filtering work will probably modify many of the same parts of the code base, so we'll need to pay attention if we want to do those in parallel.

**Comment by silentninja** [comment]
> I updated the format of the frontend, backend, design, and product sections to include summaries of completed work. Any thoughts on that?

I find it to be useful. Looking back on what we have accomplished is quite motivating and great to know the pace we have been moving on. But I would cut it short if it takes too much time. 

>  How did your most recent work go?

Error handling PR took longer than expected as I spent time looking at an alternate solution that didn't seem to work out, but I feel good about the outcome. 

> What are you planning to work on soon?

Currently looking to make any review changes that might come up with regards to the Error handling issue. Also looking to knock off issues related to date-time and column index -> id. Might get my hands on the filtering and grouping issues if I get time.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None, but would be helpful to get some quick reviews on the error handling PR as it has been blocking the frontend work for a long time

**Reply by kgodey** [reply]
I think prioritizing the column index --> id issue after error handling would be great since it also unblocks foreign key API work.
